extensions/amp-list/0.1/amp-list.js

```diff
@@ -546,8 +588,27 @@ export class AmpList extends AMP.BaseElement {
     if (!elementSrc) {
       return Promise.resolve();
     }
+
     let fetch;
-    if (this.ssrTemplateHelper_.isEnabled()) {
+    if (this.isAmpStateSrc_(elementSrc)) {
+      if (this.ssrTemplateHelper_.isEnabled()) {
+        user().error(TAG, '"amp-state" URIs cannot be used in SSR mode.');
+        return Promise.resolve();
+      }
```

Comment:
Might be a bit more readable if we did some rearranging:

```js
const isLocalSrc = this.isAmpStateSrc_(src);

if (this.ssrTemplateHelper_.isEnabled()) {
  userAssert(!isLocalSrc, ...);
  ...
} else {
  if (isLocalSrc) { // Local state.
    ...
  } else { // Remote URL.
    ...
  }
}
```

Reply:
I was also thinking its beginning to get messy...I'll do this refactor here, but I'm also going to make a bigger refactor in a followup PR after landing this.

Reply:
eh, nevermind. I'll do it here.